### PR TITLE
Comment dragging

### PR DESCRIPTION
**Pull request by gbalint**
**Problem:**
Comment dragging.

https://github.com/concrete-utopia/utopia/assets/127662/e6c3ff82-5b9e-4e46-8441-fb693e9a2fd1

**Fix:**
- added useDragging hook to implement comment dragging
- new dragged position is updated on liveblocks on mouse up
- added new field to CommentMode, so during dragging we don't show the comment mode cursor

### CODE REVIEW

**Review by ruggi — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/controls/comment-indicator.tsx [hunk 1]
```diff
@@ -32,98 +41,185 @@ export const CommentIndicator = React.memo(() => {
...
+      window.addEventListener('mousemove', onMouseMove)
+      window.addEventListener('mouseup', onMouseUp)
+    },
+    [onMouseMove, onMouseUp, dispatch],
+  )
+
+  return { onMouseDown, dragPosition }
+}
+
+function windowPositionToCanvasPosition(
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/comment-indicator.tsx:212]: This doesn't feel like it should be in here and I'm a bit suspicious of those `-10` values!

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/editor/editor-modes.ts [hunk 2]
```diff
@@ -176,10 +177,11 @@ export const EditorModes = {
       selectOnFocus: selectOnFocus,
     }
   },
-  commentMode: function (location: CanvasPoint | null): CommentMode {
+  commentMode: function (location: CanvasPoint | null, isDragging: boolean): CommentMode {
```
Comment (balazsbajorics) [inline editor/src/components/editor/editor-modes.ts:180]: nit: isDragging could be of type `'dragging' | 'not-dragging'`

File: editor/src/components/canvas/controls/comment-indicator.tsx (hunk 1 above)
Comment (gbalint) [reply editor/src/components/canvas/controls/comment-indicator.tsx]: It is not a general function, it is just about offsetting the position so the center of the indicator is at the mouse not the top/left corner. I cleaned this up a bit in fb021103202f6e1734952b497dc1a0c7eacb2026

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/editor/editor-modes.ts (hunk 2 above)
Comment (gbalint) [reply editor/src/components/editor/editor-modes.ts]: ok, see 323f73f197942d609b6bc82dd5826be5b741990c

**Pull request merged**